src/main/java/com/glencoesoftware/omero/ms/image/region/ShapeMaskRequestHandler.java

```diff
@@ -213,6 +250,66 @@ public ShapeMaskRequestHandler(ShapeMaskCtx shapeMaskCtx) {
         }
     }
 
+    public byte[] renderLabelImage(omero.client client) {
+        log.info(labelImagePath);
+        long shapeId = 156193;
+        String fullLabelImagePath = labelImagePath + Long.toString(shapeId) + ".tiledb/0/labels/StarDist/0";
+        log.info(fullLabelImagePath);
+        Array array = null;
+        try {
+            Context ctx = new Context();
+            array = new Array(ctx, fullLabelImagePath, QueryType.TILEDB_READ);
+            getData(array, ctx);
+
+        } catch (TileDBError e) {
+            log.error("Caught TileDBError", e);
+        } finally {
+            if (array != null) {
+                log.info("Closing array");
+                array.close();
+            }
+        }
+        /*
+        ArraySchema schema = array.getSchema();
+        Domain domain = schema.getDomain();
+        Attribute attribute = schema.getAttribute("a1");
+        FilterList filterList = attribute.getFilterList();)
+        */
+        return null;
+    }
+
+    private byte[] getData(Array array, Context ctx) throws TileDBError {
+        ArraySchema schema = array.getSchema();
+        Domain domain = schema.getDomain();
+        long num_dims = domain.getNDim();
+        long size = 1;
+        Dimension d0 = domain.getDimension(0);
+        log.info("Num Dims: " + Long.toString(num_dims));
+        log.info(Long.toString(domain.getRank()));
+        log.info(d0.getType().toString());
+        log.info(d0.domainToStr());
+        Pair<Byte, Byte> dimDomain = d0.getDomain();
+        for (int i = 0; i < num_dims; i++) {
+            log.info(domain.getDimension(i).domainToStr());
+        }
+        Attribute attribute = schema.getAttribute("a1");
+        log.info(attribute.getType().toString());
+        int[] subarrayDomain = new int[(int) num_dims*2];
+        for(int i = 0; i < num_dims; i++) {
+            subarrayDomain[i*2] = (int) (domain.getDimension(i).getDomain().getFirst());
+            subarrayDomain[i*2 + 1] = (int) domain.getDimension(i).getDomain().getSecond();
+        }
+        log.info(Arrays.toString(subarrayDomain));
+        NativeArray subArray = new NativeArray(ctx, subarrayDomain, Integer.class);
```

Comment:
Use  `java.nio.ByteBuffer`. See example usage in the TileDB bioformats2raw work:

* glencoesoftware/bioformats2raw#57